src/main/java/io/r2dbc/postgresql/codec/HStoreCodec.java

```diff
@@ -163,9 +162,10 @@ private static byte peekByte(ByteBuf buffer) {
     }
 
     @Nullable
-    private static String readString(StringBuilder mutableBuffer, ByteBuf buffer) {
-        mutableBuffer.setLength(0);
+    private static String readString(ByteBuf buffer) {
+        ByteBuffer accBuffer = ByteBuffer.allocate(buffer.capacity());
```

Comment:
This isn't necessary as a `ByteBuf` can be converted in to `String` by using `byteBuf.toString(StandardCharsets.UTF_8)`. We should not allocate arbitrary buffers but rather use `buffer.alloc().buffer(capacity)` to reuse the allocator and potentially pooled buffers. Make sure to `release` it after usage.

Reply:
Awesome, thank you @mp911de! To be honest, I wasn't sure how to allocate a `ByteBuf`. This also helped to make the code a little simpler! I also added a few `final` in the local variables to avoid state mutation, let me know if that's ok 🙂 

I still have a doubt about the allocation, what would be the difference if we allocate the `accBuffer` with no initial capacity?
```java
ByteBuf accBuffer = buffer.alloc().buffer();
```

Tests still pass, so I assume the buffer will start with no initial capacity and grow as required, right? Would that be better or worse for performance? 🤔